Panic when calling a function from another package where the package name alias a local variable name

CX version 0.5.18
The issue only occurs if  the package is not imported.
```
package issue
func test(i i32)() {
    printf("i : %d\n", i)
}

package main
// import "issue"
func main()() {
    var issue i32
    issue.test(issue)
}
```
output:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x781ad5]

goroutine 1 [running, locked to thread]:
github.com/skycoin/cx/cxgo/actions.ProcessMethodCall(0xc4200a8d20, 0xc420059068, 0xc420059048, 0xc4201ce001)
	/home/reash/go/src/github.com/skycoin/cx/cxgo/actions/functions.go:593 +0x545
github.com/skycoin/cx/cxgo/actions.FunctionDeclaration(0xc4201ce000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc42004a0b0, 0x2, 0x2)
	/home/reash/go/src/github.com/skycoin/cx/cxgo/actions/functions.go:95 +0x2ed
main.yyParse(0x976a60, 0xc42009c0f0, 0xc42009c0f0)
	/home/reash/go/src/github.com/skycoin/cx/cxgo/cxgo.go:1718 +0x1d09
main.main()
	/home/reash/go/src/github.com/skycoin/cx/cxgo/cxgo.nn.go:8584 +0x1640
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,10 @@
 # Changelog
 
-### v0.6.1 (NOT YET RELEASED)
+### v0.6.2 (NOT YET RELEASED)
+* Fixed issues
+  * #166: Panic when calling a function from another package where the package name alias a local variable name
+
+### v0.6.1
 * Additions
   * Support for lexical scoping
   * `if/elseif` and `if/elseif/else` statements now work properly.

--- a/cxgo/actions/functions.go
+++ b/cxgo/actions/functions.go
@@ -750,7 +750,8 @@ func ProcessMethodCall(expr *CXExpression, symbols *[]map[string]*CXArgument, of
 				}
 				argOut, err := lookupSymbol(out.Package.Name, out.Name, symbols)
 				if err != nil {
-					panic("")
+					println(CompilationError(out.FileName, out.FileLine), fmt.Sprintf("identifier '%s' does not exist", out.Name))
+					os.Exit(CX_COMPILATION_ERROR)
 				}
 				// then we found an output
 				if len(out.Fields) > 0 {
@@ -796,6 +797,11 @@ func ProcessMethodCall(expr *CXExpression, symbols *[]map[string]*CXArgument, of
 
 					strct := argOut.CustomType
 
+					if strct == nil {
+						println(CompilationError(argOut.FileName, argOut.FileLine), fmt.Sprintf("illegal method call or field access on identifier '%s' of primitive type '%s'", argOut.Name, TypeNames[argOut.Type]))
+						os.Exit(CX_COMPILATION_ERROR)
+					}
+
 					expr.Inputs = append(expr.Outputs[:1], expr.Inputs...)
 
 					expr.Outputs = expr.Outputs[:len(expr.Outputs)-1]
@@ -816,13 +822,18 @@ func ProcessMethodCall(expr *CXExpression, symbols *[]map[string]*CXArgument, of
 
 			argOut, err := lookupSymbol(out.Package.Name, out.Name, symbols)
 			if err != nil {
-				println(CompilationError(out.FileName, out.FileLine), fmt.Sprintf("'%s.%s' not found", out.Package.Name, out.Name))
+				println(CompilationError(out.FileName, out.FileLine), fmt.Sprintf("identifier '%s' does not exist", out.Name))
 				os.Exit(CX_COMPILATION_ERROR)
 			}
 
 			// then we found an output
 			if len(out.Fields) > 0 {
 				strct := argOut.CustomType
+
+				if strct == nil {
+					println(CompilationError(argOut.FileName, argOut.FileLine), fmt.Sprintf("illegal method call or field access on identifier '%s' of primitive type '%s'", argOut.Name, TypeNames[argOut.Type]))
+					os.Exit(CX_COMPILATION_ERROR)
+				}
 
 				if fn, err := strct.Package.GetMethod(strct.Name+"."+out.Fields[len(out.Fields)-1].Name, strct.Name); err == nil {
 					expr.Operator = fn

--- a/tests/main.cx
+++ b/tests/main.cx
@@ -403,7 +403,7 @@ func main ()() {
 	runTest("issue-156.cx", cx.COMPILATION_ERROR, "Panic when using a function declared in another package without importing the package")
 	runTest("issue-157.cx", cx.SUCCESS, "Cx memory stomped")
 	runTest("issue-158.cx", cx.SUCCESS, "Invalid offset calculation of non literal strings when appended to a slice")
-	runTestEx("issue-166.cx", cx.SUCCESS, "Panic when calling a function from another package where the package name alias a local variable name", TEST_ISSUE, 0)
+	runTest("issue-166.cx", cx.COMPILATION_ERROR, "Panic when calling a function from another package where the package name alias a local variable name")
 	runTest("issue-167.cx", cx.SUCCESS, "Garbage memory when passing the address of slice element to a function")
 	runTest("issue-168.cx", cx.SUCCESS, "Type deduction of struct field fails")
 	runTest("issue-169.cx", cx.COMPILATION_ERROR, "No compilation error when assigning a i32 value to a []i32 variable")

--- a/tests/testdata/tokens/main.cx.txt
+++ b/tests/testdata/tokens/main.cx.txt
@@ -3170,19 +3170,15 @@ PERIOD
 STRLIT Invalid offset calculation of non literal strings when appended to a slice
 RPAREN 
 SCOLON 
- IDENT runTestEx
+ IDENT runTest
 LPAREN 
 STRLIT issue-166.cx
  COMMA 
  IDENT cx
 PERIOD 
- IDENT SUCCESS
+ IDENT COMPILATION_ERROR
  COMMA 
 STRLIT Panic when calling a function from another package where the package name alias a local variable name
- COMMA 
- IDENT TEST_ISSUE
- COMMA 
-INTLIT 0
 RPAREN 
 SCOLON 
  IDENT runTest

